### ISSUE TITLE
golint incorrectly reports that an exported constant/variable is uncommented 

### ISSUE THREAD

**Issue by opennota** [golang/lint#208]
test.go:
``` go
package test

const Test = 1 // comment
```

```
test.go:3:7: exported const Test should have comment or be unexported
```

And if you replace `const` with `var`:

```
test.go:3:5: exported var Test should have comment or be unexported
```

**Comment by corburn** [comment]
```
// Test should not trigger a lint warning if the comment immediately precedes the variable.
const Test = 1
```

**Comment by dsymonds** [comment]
Yep, what @corburn said. It's mandated that way because that's the form that godoc extracts.

**Comment by opennota** [comment]
I must differ, because godoc.org shows these comments just fine.

**Comment by opennota** [comment]
https://godoc.org/github.com/opennota/elvenshtein

**Comment by opennota** [comment]
At the very least, I'd like golint not to tell me outright lies, please.

**Comment by opennota** [comment]
Please reopen.

**Comment by morriswinkler-simplesurance** [comment]
+1
it uglifies blocks of constants
